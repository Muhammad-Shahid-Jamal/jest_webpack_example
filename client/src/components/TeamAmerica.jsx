import React from 'react';
import { teams } from './team.js';

export default function TeamAmerica() {
  return (
    <div>
      <h1>Team America</h1>
      {
        teams.map((team, i) => (
          <div key={i}>
            <h2>{team[0].team}</h2>
            {team.map((member, k) => {
              return (
                <ul key={k}>
                  <li>
                    {member.full_name}&nbsp;-&nbsp;
                    {member.role}&nbsp;
                  </li>
                </ul>
              );
            })}
          </div>
        ))
      }
    </div>
  );
}