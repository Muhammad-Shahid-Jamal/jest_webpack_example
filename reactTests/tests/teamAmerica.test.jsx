import React from 'react';
import renderer from 'react-test-renderer';
import TeamAmerica from '../../client/src/components/TeamAmerica';

// This is the actual test written for Jest
it('test to see if the team renders correctly', () => {
  const tree = renderer.create(<TeamAmerica />).toJSON();
  expect(tree).toMatchSnapshot();
});

