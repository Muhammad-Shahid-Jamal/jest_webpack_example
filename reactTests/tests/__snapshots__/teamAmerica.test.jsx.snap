// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the team renders correctly 1`] = `
<div>
  <h1>
    Team America
  </h1>
  <div>
    <h2>
      Presidents
    </h2>
    <ul>
      <li>
        George Washington
         - 
        First President of the United State of America.
         
      </li>
    </ul>
    <ul>
      <li>
        Abraham Lincoln
         - 
        President during the Civil War.
         
      </li>
    </ul>
    <ul>
      <li>
        Franklin D. Roosevelt
         - 
        President during The Great Depression and WW2. Only president to be elected 4 times.
         
      </li>
    </ul>
  </div>
  <div>
    <h2>
      US People
    </h2>
    <ul>
      <li>
        Joe Sixpack
         - 
        Average American
         
      </li>
    </ul>
  </div>
</div>
`;
